### DISCUSSION TITLE
Is iso8601 long format intended to show a 2 digit year?

### DISCUSSION THREAD

**Discussion by git-help-eng** [jonaswinkler/paperless-ng#729] [Archived FRs]
I'm running version 1.3.0 in docker and my iso8601 long date is not showing how it seems that it should be.

I have selected the long iso8601 format:
![selected long format](https://user-images.githubusercontent.com/79425928/110417346-6cc68800-8063-11eb-9ea9-45c9464d7e4d.png)

But in the documents window I see the year with only 2 digits as seen below:
![shown date format](https://user-images.githubusercontent.com/79425928/110417380-7e0f9480-8063-11eb-92f9-2ada34e71593.png)

I thought it might be a design choice due to the limited width of the container, but selecting the long dutch time fomat (with the same number of characters) shows that it does not seem to be a width constraint as shown below:
![dutch long format](https://user-images.githubusercontent.com/79425928/110417498-b7480480-8063-11eb-9813-c9f401d178ce.png)

Is anyone else seeing this behaviour?

**Comment by jonaswinkler** [comment]
Well yes, this behavior is intended - the small cards always use the shortest date format available in the current locale, and I've manually defined the short ISO-8601 option to only show two digits for the year. I recall that I did have some issues with the full format (yyyy-mm-dd). The exact width of the date is not consistent across browsers (different fonts, different screen DPI, etc), therefore I went with the safe option.

Not sure why the Dutch short date format shows 4 digit years, whereas pretty much all the other options use two digits.

**Comment by git-help-eng** [comment]
@jonaswinkler,
    Thanks for the reply.  I did evaluate which selections from the "Date display" dropdown show a 4 digit year (5 of them do).  Whether they do appears to be based on the number of digits in the year displayed in the dropdown list.
![out](https://user-images.githubusercontent.com/79425928/110654260-fc675600-818b-11eb-8c69-fa454e870590.png)

If you do make any changes to change/harmonize that behaviour in the future it is my opinion that ISO8601 should use a 4 digit year, per the standard.  I fully understand why it is not that way but just wanted to give my opinion.

Thanks for effort that you have put into this project.  The new interface makes organizing and harmonizing documents so much easier.

**Reply by jonaswinkler** [reply]
> it is my opinion that ISO8601 should use a 4 digit year, per the standard.

Agreed, it's a standard after all. I'll change that.

The localized date formats are provided by the development framework I'm using and outside of my control.

**Comment by git-help-eng** [comment]
@jonaswinkler,
    Thanks for the change.
